test/lib/index.test.js

```diff
@@ -277,4 +277,47 @@ describe( 'lib/index', function() {
             expect( result3.error ).to.exist;
         });
     });
+
+    describe("Joi schema extended with custom type", function() {
+        let parser;
+
+        const customJoi = joi.extend(joi => ({
+            base: joi.string(),
+            name: "string20",
+            language: {
+                valid: "must be string with less than 20 characters",
+            },
+            rules: [
+                {
+                    name: "valid",
+                    description: "Valid String 20",
+                    validate(params, value, state, options) {
+                        if (value.length >= 20) {
+                            return this.createError("string20.valid", {}, state, options);
+                        }
+                        return value;
+                    },
+                },
+            ],
+        }));
+
+        beforeEach(function() {
+            parser = index.parser(customJoi);
+        });
+
+        it("custom type valid and required", function() {
+            const schema = parser.parse({
+                "@type": "string20",
+                // use null to call with no params.
+                valid: null,
```

Comment:
This feels unintuitive. Was it original functionality that you're just replicating?

Reply:
Which part being unintiutive - using null to tell the system to call the function without an argument?

If so, that's part of the original functionality. `lib/base.js updateSchema()`

Reply:
Yes, sounds good then. :+1: